core/arch/arm/plat-imx/sub.mk

```diff
@@ -4,7 +4,8 @@ srcs-y += main.c
 srcs-$(CFG_PL310) += imx_pl310.c
 srcs-$(CFG_PSCI_ARM32) += psci.c
 
-srcs-$(PLATFORM_FLAVOR_mx6qsabrelite) += a9_plat_init.S
-srcs-$(PLATFORM_FLAVOR_mx6qsabresd) += a9_plat_init.S
-srcs-$(PLATFORM_FLAVOR_mx6dlsabresd) += a9_plat_init.S
-srcs-$(PLATFORM_FLAVOR_mx6ulevk) += imx6ul.c
+ifneq (,$(filter y, $(CFG_MX6Q) $(CFG_MX6D) $(CFG_MX6DL)))
+srcs-y += a9_plat_init.S imx6.c
```

Comment:
**imx6.c** sounds strange when a **mx6ul** is excluded from its support.
**imx6.c** could be renamed **imx6-a9.c**.